libs/model/src/user/SignIn.command.ts

```diff
@@ -0,0 +1,209 @@
+        const auth = await verifyAddress(community_id, address.trim());
+        // return the signed in user or a dummy placeholder
+        return {
+          id: signedInUser?.id ?? -1,
+          email: signedInUser?.email ?? '',
+          auth,
+        };
+      },
+    },
+    body: async ({ actor, payload }) => {
+      if (!actor.user.auth) throw Error('Invalid address');
+
+      const { community_id, wallet_id, referrer_address, session, block_info } =
+        payload;
+      const { base, encodedAddress, ss58Prefix, hex, existingHexUserId } = actor
+        .user.auth as VerifiedAddress;
+
+      let user_id =
+        (actor.user?.id ?? 0) > 0 ? actor.user.id : (existingHexUserId ?? null);
+
+      await verifySessionSignature(
+        deserializeCanvas(session),
+        encodedAddress,
+        ss58Prefix,
+      );
+
+      const verification_token = crypto.randomBytes(18).toString('hex');
+      const verification_token_expires = new Date(
+        +new Date() + config.AUTH.ADDRESS_TOKEN_EXPIRES_IN * 60 * 1000,
+      );
+
+      // upsert address, passing user_id if signed in
+      const { user_created, address_created, first_community } =
+        await models.sequelize.transaction(async (transaction) => {
+          // is same address found in a different community?
+          const is_first_community = !(
+            user_id ||
+            (await models.Address.findOne({
+              where: {
+                community_id: { [Op.ne]: community_id },
+                address: encodedAddress,
+              },
+              attributes: ['community_id'],
+              transaction,
+            }))
+          );
+
+          /* If address doesn't have an associated user, create one!
+          - NOTE: magic strategy is the other place (when using email)
+          */
+          let new_user = false;
+          if (!user_id) {
+            const existing = await models.Address.findOne({
+              where: { address: encodedAddress, user_id: { [Op.ne]: null } },
+            });
+            if (!existing) {
+              const user = await models.User.create(
+                { email: null, profile: {} },
+                { transaction },
+              );
+              if (!user) throw new Error('Failed to create user');
+              user_id = user.id!;
+              new_user = true;
+            } else user_id = existing.user_id!;
+          }
+
+          const [addr, new_address] = await models.Address.findOrCreate({
+            where: { community_id, address: encodedAddress },
+            defaults: {
+              community_id,
+              address: encodedAddress,
+              user_id,
+              hex,
+              wallet_id,
+              verification_token,
+              verification_token_expires,
+              block_info,
+              last_active: new Date(),
+              verified: new Date(),
+              role: 'member',
```

Comment:
Won't this cause issues for admin sign in?

Reply:
Are admins created as admins the first time? or created as members and then upgraded to admin via a different command?

Reply:
resolving this to merge